### PR TITLE
added first version of hover functionality

### LINKED ISSUE #48: Switch detail view when hovering over a confmat cell
Similar to http://ml4a.github.io/demos/confusion_cifar/

### PR DESCRIPTION
**Pull request by gfrogat**
Closes #48.

the variable names are probably a bit off but I noticed that we're generally not than consistent throughout the app and use label or class interchangeably ...

### CODE REVIEW

File: src/confusion_matrix_cell/ACellRenderer.ts [hunk 1]
```diff
@@ -97,16 +97,30 @@ export class LineChartRenderer extends ACellRenderer {
...
+      .attr('stroke', coloring)
       .attr('stroke-opacity', '0.6')
       .append('title')
-      .text((d) => d.classLabel);
+      .text((d) => d.predictedLabel);
 
     $node.select('g').selectAll('.instance-line').attr('d', (d) => line(DataStoreApplicationProperties.switchToAbsolute ? d.values : d.valuesInPercent));
+
+    if (DataStoreApplicationProperties.isHighlighted) {
+      DataStoreApplicationProperties.toggleHighLighting();
```
Comment (thinkh) [inline src/confusion_matrix_cell/ACellRenderer.ts]: Why do you call the `toggleHighLighting()` here? What does it do?

Comment (thinkh) [reply src/confusion_matrix_cell/ACellRenderer.ts]: Solved this with an `mouseout` listener

**Comment by thinkh** [comment]
I refactored the code and added a mouseout listener. This way the highlighted in the detail view is cleared when the mouse hovers a cell outside the pinned row/column.

malevo-cell-hover-mouseout

**Comment by thinkh** [comment]
@gfrogat Please have a look if you agree with the current behavior. If yes, you can go ahead and merge this PR.

**Comment by gfrogat** [comment]
Looks great! Thanks a lot for the cleanup!

Regarding the `mouseout` event: I had this in my code too at one point, but I was not sure whether it's better to always reset the detail chart or to have a way to keep the current "hover-selection" active (e.g. for when using different scalings) by moving out into an "independent" row/column. 

Now the user always have to go forward and backward between the cell and the sliders.

Maybe we could bind the scaling to the mouse-wheel?

**Comment by thinkh** [comment]
> Regarding the mouseout event: I had this in my code too at one point, but I was not sure whether it's better to always reset the detail chart or to have a way to keep the current "hover-selection" active (e.g. when using different scalings) by moving out into an "independent" row/column.

I thought of this, too. However, I find it more consistent now. However, I'm still not happy with the current solution, as I'm hovering "blind" when I focus on the detail chart. Because focusing on the mouse cursor and the detail chart at the same time is hard. ;-)

> Maybe we could bind the scaling to the mouse-wheel?

I'm not so sure, if this is a good idea. We can only do this, if we can ensure that the layout will always fit into the viewport (without vertical scrollbar).

> Update: Or we just keep it as it is now and change it if somebody complains

I would keep it like this for now. We can change/modify it later if we have a better idea. And maybe it works like this for users.

**Comment by gfrogat** [comment]
ok, let’s merge it so we have it in the demo and discuss it in detail next week

**Pull request merged**